src/main/java/de/tum/in/www1/artemis/domain/ProgrammingSubmission.java

```diff
@@ -37,6 +39,11 @@
     @Cache(usage = CacheConcurrencyStrategy.NONSTRICT_READ_WRITE)
     private List<BuildLogEntry> buildLogEntries = new ArrayList<>();
 
+    // If the submission is deleted, the testwise coverage report with all child entities are deleted
```

Comment:
please add a comment that this is only valid for programming submissions related to template or solution participations

Reply:
In fact, it is only related to a solution participation. See an explanation [here](https://github.com/ls1intum/Artemis/pull/4889#discussion_r842775232).

I added some JavaDoc.